tools/proc_init.py

```diff
@@ -0,0 +1,302 @@
+    '''
+    recv_tensor_shape = torch.zeros(dimensions, dtype = torch.int32)
+    dist.recv(recv_tensor_shape, src=0)
+    recv_shape = list( map( lambda x: int(x), recv_tensor_shape) )
+
+    '''
+    Receive the data message here for nodes here. 
+    '''
+    recv_tensor_data = torch.zeros( recv_shape, dtype=dtype)
+    dist.recv( recv_tensor_data, src=0 )
+    return recv_tensor_data.numpy ()
+
+'''
+Function to receive node data
+'''
+def recvNodeData( rank, dimensions, dtype ): 
+    return recvData( rank, dimensions, dtype )
+
+'''
+Invoke the Node Data here as well. it will serve the purpose. 
+'''
+def recvEdgeData (rank, dimensions, dtype): 
+    return recvNodeData (dimensions, dtype)
+
+'''
+This function is for receiving node features. 
+Note that the node features are indexed by the orig_node_type_id. 
+for the mag-dataset, only node_type = 3 has node features and others node types does not
+have any node features. 
+'''
+def recvNodeFeatures (rank, dtype): 
+    globalIds = recvNodeData( rank, 1, torch.int32 ) ## to receive the origTypeNodeIDs
+    node_features = recvNodeData( rank, 2, torch.float32 ) ## to receive actual node feature data
+    print( 'Rank: ', rank, ', Done receiving node feature data... ' )
+    return globalIds, node_features
+
+def recv_node_features_obj(rank, size): 
+    send_objs = [None for _ in range(size)]
+    recv_obj = [None]
+    dist.scatter_object_list(recv_obj, send_objs, src=0)
+
+    node_features = recv_obj[0]
+    return node_features
+
+
+def readInputFiles( rank, params, metis_partitions ): 
+
+    '''
+    Node data is structured as follows: 
+    <node_type> <weight1> <weight2> <weight3> <weight4> <orig_type_node_id> <attributes>
+    is converted to 
+    <node_type> <weight1> <weight2> <weight3> <weight4> <orig_type_node_id> <nid> <recv_proc>
+    '''
+    rcvProc_nodes_data = []
+    nodes_data = read_nodes_file( params.input_dir+'/'+params.nodes_file )
+    rcvProc_nodes_data = include_recv_proc_nodes( nodes_data, metis_partitions )
+    print( 'Rank: ', rank, ', Completed loading nodes data: ', rcvProc_nodes_data.shape )
+
+    rcvProc_edge_data = []
+    edges_data = read_edge_file( params.input_dir+'/'+params.edges_file )
+    rcvProc_edge_data = include_recv_proc_edges( edges_data, metis_partitions )
+    print( 'Rank: ', rank, ', Completed loading edges data: ', rcvProc_edge_data.shape )
+
+    node_features = []
+    node_features = read_node_features_file( params.input_dir+'/'+params.node_feats_file )
+    print( 'Rank: ', rank, ', Completed loading node features reading from file ', len(node_features))
+
+    edge_features = []
+    #edge_features = read_edge_features_file( params.input_dir+'/'+params.edge_feats_file )
+    #print( 'Rank: ', rank, ', Completed edge features reading from file ', len(edge_features) )
+
+    return rcvProc_nodes_data, node_features, rcvProc_edge_data, edge_features
+
+def run(rank, size, params):
```

Comment:
this name is too general. if this is a local function, the name might be fine.

Reply:
i think it's better to move this function to the script that runs the command line.